WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -2,7 +2,7 @@
 
 class Api::V0::UserRolesController < Api::V0::ApiController
   include SortHelper
-  before_action :current_user_is_authorized_for_action!, only: [:create, :update, :destroy]
+  before_action :current_user_is_authorized_for_action!, only: [:update, :destroy]
```

Comment:
I don't quite get why you are replacing one `before_action` check by three relatively verbose `return 401` checks inline below. Can you explain?

Reply:
It's because the permission check needs the request params as well. Eg: some people might have create access for delegate_regions, but not for a team. The same is applicable for update and destroy as well and will change that as well in future.